Create demonstrations.rst (#96)
LSSTDESC/RAIL: docs/source/demonstrations.rst

*****************
Example Notebooks
*****************

RAIL comes with several notebooks that demonstrate how to use it to analyze data in a number of different ways.

> Here we describe the various notebooks and suggest and other in which you might study them.

Reviewer: ```suggestion
Here we describe the various notebooks and suggest other ways in which you might study the data.
```

Starting out, overview notebooks
================================

We recommend starting with the `Goldenspike <https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/goldenspike_notebook.html>`_ notebook, 
which demonstrates a relatively simple end-to-end analysis.  This analysis starts off by making a model that can be used to generate synthetic 
catalogs of photometric data.  It then uses that model to create sets of synthetic data to train and test per-object redshift estimators, i.e., 
estimators that compute p(z).  From there it trains and tests a few estimators using some common algorithms.   It then evaluates the 
performance of those estimators.   Finally, it shows a few methods that converts p(z) for a set of objects to an ensemble distribution n(z).

The `estimation <https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/rendered/estimation_examples/RAIL_estimation_demo.html>`_ notebook 

> focus more on the estimation parts of the analysis, and demonstrates a few additional estimation algorithms.

Reviewer: ```suggestion
focuses more on the estimation parts of the analysis, and demonstrates a few additional estimation algorithms.
```

The `evaluation <https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/rendered/evaluation_examples/demo.html>`_ of the estimator performance is described in more depth in its own notebook.

> Finally, we have collected demonstrates of `useful utilites <https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/rendered/core_examples/Useful_Utilities.html>`_ to explore which packages and algorithms are available in the current RAIL installation.

Reviewer: ```suggestion
Finally, we have collected demonstrations of `useful utilites <https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/rendered/core_examples/Useful_Utilities.html>`_ to explore which packages and algorithms are available in the current RAIL installation.
```

Deeper dives into synthetic data creation
=========================================

> The notebooks in  `this <https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/creation_notebooks.html>`_ directory demonstate how 

Reviewer: ```suggestion
The notebooks in  `this <https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/creation_notebooks.html>`_ directory demonstrate how 
```

> how to generate synthetic photometric data, and also how "degrade" the synthetic data by applying various effects to the data.

Reviewer: ```suggestion
how to generate synthetic photometric data, and also how to "degrade" the synthetic data by applying various effects to the data.
```

<https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/rendered/estimation_examples/NZDir.html>`_ algorithms.

These two notebooks demonstrate self-organizing map (SOM) based algorithms that estimate the ensemble n(z) distribution: `the first  <https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/rendered/estimation_examples/somocluSOM_demo.html>`_ works with the SOM directly, 
while `the second <https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/rendered/estimation_examples/somocluSOMcluster_demo.html>`_ clusters the SOM cells to reduce statistical fluctuations.

Finally, `this notebook <https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/rendered/estimation_examples/test_sampled_summarizers.html>`_ demonstrates converting collections of per-object p(z) estimates to ensemble n(z) estimates.

Deeper explanations of rail concepts
====================================

The two notebooks demonstrate how to `convert a notebook into a ceci analysis pipeline 

> <https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/rendered/core_examples/Build_Save_Load_Run_Pipeline.html>`_ and how 

Reviewer: ```suggestion
<https://rail-hub.readthedocs.io/projects/rail-notebooks/en/latest/rendered/core_examples/Build_Save_Load_Run_Pipeline.html>`_ and how to
```
